### PR TITLE
ci: disable upstream trigger on PRs for the packaging job

### PR DESCRIPTION
**Pull request by kuisathaverat**
## What does this PR do?

* Change to use withMageEnv
* Disable upstream trigger on PRs
* Disable macOS sign packages is a bottleneck

## Why is it important?

This PR fixes the queue-boom occasioned by the upstream trigger on PRs.

## Checklist

- ~~[ ] My code follows the style guidelines of this project~~
- ~~[ ] I have commented my code, particularly in hard-to-understand areas~~
- ~~[ ] I have made corresponding changes to the documentation~~
- ~~[ ] I have made corresponding change to the default configuration files~~
- ~~[ ] I have added tests that prove my fix is effective or that my feature works~~
- ~~[ ] I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

### CODE REVIEW

File: .ci/packaging.groovy [hunk 1]
```diff
@@ -23,10 +23,10 @@ pipeline {
   }
   triggers {
     issueCommentTrigger('(?i)^\\/packag[ing|e]$')
-    upstream('Beats/beats-beats-mbp/master')
+    upstream("Beats/beats-beats-mbp/${ env.JOB_BASE_NAME.startsWith('PR-') ? 'none' : env.JOB_BASE_NAME }")
```
Comment (v1v) [inline .ci/packaging.groovy:26]: I'd say to add a comment with the reason too:

```suggestion
    // disable upstream trigger on a PR basis
    upstream("Beats/beats-beats-mbp/${ env.JOB_BASE_NAME.startsWith('PR-') ? 'none' : env.JOB_BASE_NAME }")
```

File: .ci/packaging.groovy [hunk 2]
```diff
@@ -216,38 +216,18 @@ def publishPackages(baseDir){
...
-      sh(label: "Install Go ${GO_VERSION}", script: ".ci/scripts/install-go.sh")
-      sh(label: "Install Mage", script: "make mage")
-      body()
+  withMageEnv(){
+    withEnv([
+      "PYTHON_ENV=${WORKSPACE}/python-env"
+    ]) {
+      deleteDir()
+      //unstash 'source'
+      gitCheckout(basedir: "${BASE_DIR}")
```
Comment (v1v) [inline .ci/packaging.groovy:225]: Do you want to use the stashV2 and unstashV2 steps rather than gitCheckout or stash/unstash?

Comment (kuisathaverat) [reply .ci/packaging.groovy:225]: sure in a follow up

File: .ci/packaging.groovy [hunk 3]
```diff
@@ -216,38 +217,18 @@ def publishPackages(baseDir){
...
+      dir("${env.BASE_DIR}"){
+        sh(label: "Install Go ${GO_VERSION}", script: ".ci/scripts/install-go.sh")
```
Comment (mdelapenya) [inline .ci/packaging.groovy:228]: Wdyt about having this install-go script bundled into the library? We are using a script from travis elsewhere

Comment (kuisathaverat) [reply .ci/packaging.groovy:228]: indeed it is withMageEnv, this line should be removed

**Pull request merged**